algebird-shapeless/src/main/scala/com/twitter/algebird/shapeless/HListSum.scala

```diff
@@ -0,0 +1,14 @@
+package com.twitter.algebird.shapeless
+
+import shapeless._
+import shapeless.ops.hlist.ToList
+import com.twitter.algebird._
+
+object HListSum {
+  def sum[L <: HList, T](hlist: L)(implicit toList: ToList[L, T], m: Monoid[T]): T =
+    m.sum(toList(hlist))
+
+  def sum1[L <: HList, T](cons: T :: L)(implicit toList: ToList[L, T], s: Semigroup[T]): T =
```

Comment:
I find the following little confusing. Given:

```scala
import com.twitter.algebird.Monoid
import shapeless.{ ::, HNil }

abstract class Base { def x: Int }
case class Foo(x: Int) extends Base
object Base {
  implicit val baseMonoid: Monoid[Base] =
    Monoid.from[Base](Foo(0))((a, b) => Foo(a.x + b.x))
}
```
And then:
```scala
scala> import com.twitter.algebird.shapeless.HListSum
import com.twitter.algebird.shapeless.HListSum

scala> val l = Foo(1) :: (Foo(2): Base) :: HNil
l: shapeless.::[Foo,shapeless.::[Base,shapeless.HNil]] = Foo(1) :: Foo(2) :: HNil

scala> HListSum.sum(l)
res0: this.Lub = Foo(3)

scala> HListSum.sum1(l)
<console>:16: error: could not find implicit value for parameter toList: shapeless.ops.hlist.ToList[shapeless.::[Base,shapeless.HNil],Foo]
       HListSum.sum1(l)
                    ^
```
I _think_ switching the order of the type parameters would fix it (?).

Reply:
how did you even discover that?

I'll add a test and try to fix it.